themes/default/content/challenge/holiday-shopping.md

```diff
@@ -0,0 +1,269 @@
+
+```python
+# An ECS cluster to deploy into
+cluster = aws.ecs.Cluster("cluster")
+
+# An ALB to serve the frontend service to the internet
+lb = awsx.lb.ApplicationLoadBalancer("grocery-lb")
+```
+
+Now, create an [AWS Fargate](https://www.pulumi.com/learn/glossary/aws-fargate/) service that will run the containers. There are some configuration values for the Fargate service that you can set via Pulumi config, but for this example, you can just use the default values. This code creates a Fargate service that runs a task with two containers, a front and back. The front container is using the frontend image created in Step 1 and is wired to receive traffic from the ALB. The back container is using the backend image and listens on port 8000.
+
+```python
+# AWS configs
+container_port = config.get_int("containerPort", 80)
+cpu = config.get_int("cpu", 1024)
+memory = config.get_int("memory", 1024)
+
+# Deploy an ECS Service on Fargate to host the application containers
+service = awsx.ecs.FargateService(
+    "grocery-service",
+    cluster=cluster.arn,
+    task_definition_args=awsx.ecs.FargateServiceTaskDefinitionArgs(
+        containers={
+            "front": awsx.ecs.TaskDefinitionContainerDefinitionArgs(
+                image=frontend_image.image_uri,
+                cpu=cpu,
+                memory=memory,
+                essential=True,
+                port_mappings=[awsx.ecs.TaskDefinitionPortMappingArgs(
+                    container_port=container_port,
+                    target_group=lb.default_target_group,
+                )],
+            ),
+            "back": awsx.ecs.TaskDefinitionContainerDefinitionArgs(
+                image=backend_image.image_uri,
+                cpu=cpu,
+                memory=memory,
+                essential=True,
+                port_mappings=[awsx.ecs.TaskDefinitionPortMappingArgs(
+                    container_port=8000,
+                    host_port=8000
+                )],
+            ),
+        }  
+    ),
+    desired_count=1
+)
+
+# AWS export
+pulumi.export("app url", Output.concat("http://", lb.load_balancer.dns_name))
+```
+
+Run `pulumi up` and select yes to perform the update. Pulumi will output the `app url`. Go to the URL, and you will be presented with Pulumipus' Grocery List web application. You can try to add some items, but nothing will happen because there is no database connected yet. You will do that in the next step.
+
+#### Step 3. Create a MongoDB Atlas database
+
+The web app stores the grocery list items inside a MongoDB database. MongoDB Atlas is a fully managed cloud database that you will be using for this application. Add the below before the code from the previous steps in `__main__.py`. This code creates a MongoDB project, sets up the network access controls, creates a Free Tier cluster in `US_WEST_2` in AWS, and a database user that is scoped to the created cluster.
```

Comment:
> Add the below before the code

We should probably say exactly where this code should go, like "above the ECR repo declaration" or something, just so that's clear.